src/dmd/parse.d

```diff
@@ -1515,6 +1515,37 @@ final class Parser(AST) : Lexer
         }
     }
 
+    //Indicates if a pack of TypeCtor, as allowed before a BasicType, can be read.
+    private bool isTypeCtorPack()
+    {
+        Token* tk = &token;
+        bool result;
+        while (true)
```

Comment:
I'm sorry, but I don't consider the matter of consolidating this code with https://github.com/dlang/dmd/blob/55ca5920ca0f7da1f4096eafb95feaffa6813a79/src/dmd/parse.d#L1518-L1552 resolved.  At a minimum, the `case` statement can be put into a function shared by both implementations (perhaps somewhere near the `TOK` enum declaration).

Reply:
Dammit.  Hit the wrong button.  I mean't to select "Request Changes"

Reply:
I've managed to apply your suggestion so this is resolved for good this time